src/nvim/rbuffer.c

```diff
@@ -121,7 +121,7 @@ char *rbuffer_read_ptr(RBuffer *buf, size_t *read_count) FUNC_ATTR_NONNULL_ALL
 {
   if (!buf->size) {
     *read_count = 0;
-    return NULL;
+    return buf->read_ptr;
```

Comment:
Yes, nonzero size still needs to be checked, but some(/all) of the `NULL` checks could be removed. `out_data_append_to_screen()`  could also be changed, because this function is now never called with a `NULL` pointer for `output` (`FUNC_ATTR_NONNULL_ALL`?).